src/jpgimage.cpp

```diff
@@ -668,23 +669,23 @@ namespace Exiv2 {
                             // and dumping the XMP in a post read operation similar to kpsIptcErase
                             // for the moment, dumping 'on the fly' is working fine
                             if (!bExtXMP) {
-                                while (xmp[start])
+                                while (xmp.at(start))
                                     start++;
                                 start++;
-                                if (::strstr((char*)xmp + start, "HasExtendedXMP")) {
+                                std::string xmp_from_start = string_from_unterminated((char*)&xmp.at(start), size - start);
+                                if (xmp_from_start.find("HasExtendedXMP", start) != xmp_from_start.npos) {
                                     start = size;  // ignore this packet, we'll get on the next time around
                                     bExtXMP = true;
                                 }
                             } else {
                                 start = 2 + 35 + 32 + 4 + 4;  // Adobe Spec, p19
                             }
 
-                            out.write((const char*)(xmp + start), size - start);
-                            delete[] xmp;
+                            out.write((const char*)(&xmp.at(start)), size - start);
```

Comment:
[suggestion] Have you considered the usage of `reinterpret_cast` here?. When the compiler sees a C-style or function style cast, it tries to apply different sequences of elementary conversion. The first one that is possible is applied. The sequences are in order:

`const_cast`
`static_cast`
`static_cast` followed by `const_cast`
`reinterpret_cast`
`reinterpret_cast` followed by `const_cast`

Reply:
Would be definitely the better thing to do. I'll do that.

Reply:
Done